lib/Localization/LocalizationFormat.cpp

```diff
@@ -85,9 +91,27 @@ bool SerializedLocalizationWriter::emit(llvm::StringRef filePath) {
   return OS.has_error();
 }
 
+llvm::StringRef
+LocalizationProducer::getMessageOr(swift::DiagID id,
+                                   llvm::StringRef defaultMessage) {
+  auto localizedMessage = getMessage(id);
+  if (localizedMessage.empty())
+    return defaultMessage;
+  llvm::StringRef diagnosticName(diagnosticNameStrings[(unsigned)id]);
+  if (localizationSaver && printDiagnosticName) {
+    auto localizedDebugDiagnosticMessage =
+        localizationSaver->save(localizedMessage.str() + diagnosticName.str());
```

Comment:
The `localizationSaver` will save the strings in its allocator and since the `saver` and the `allocator` are present for the `DiagnosticEngine`'s life cycle... the strings won't be released, right?